src/main/java/org/candlepin/dto/hosted/SubscriptionTranslator.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.hosted;
+
+import org.candlepin.dto.ModelTranslator;
+import org.candlepin.dto.ObjectTranslator;
+import org.candlepin.dto.api.v1.CdnDTO;
+import org.candlepin.dto.api.v1.CertificateDTO;
+import org.candlepin.dto.api.v1.NestedOwnerDTO;
+import org.candlepin.dto.api.v1.ProductDTO;
+import org.candlepin.dto.api.v1.SubscriptionDTO;
+import org.candlepin.model.dto.ProductData;
+import org.candlepin.model.dto.Subscription;
+import org.candlepin.util.Util;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+/**
+ * The SubscriptionTranslator provides translation from Subscription objects to the
+ * SubscriptionDTO (hosted test API)
+ */
+public class SubscriptionTranslator implements ObjectTranslator<Subscription, SubscriptionDTO> {
```

Comment:
Need to add unit tests for the new translator classes

Reply:
Done.